### PR TITLE
Built my Queue

### PR DESCRIPTION
**Pull request by r3ddawn**
Built my Queue and finished adding and testing push and pop
functionality.

### CODE REVIEW

**Comment by r3ddawn** [comment]
Was going to work on Stack before Queue, but ran into Index issues. This prompted my to take a break and build Queue before returning to finish  Stack.

File: src/lists/LinkedList.java [hunk 1]
```diff
@@ -1,7 +1,7 @@
 package lists;
 
 public class LinkedList implements List {
-    private static class Node {
+    public static class Node {
```
Comment (booleangate) [inline src/lists/LinkedList.java]: You changed this to public, but you're not using this outside of this class (which you shouldn't be anyway).

File: src/lists/LinkedList.java [hunk 2]
```diff
@@ -10,7 +10,7 @@ public Node(String value) {
         }
     }
     
-    private Node head, tail;
+    public Node head, tail;
```
Comment (booleangate) [inline src/lists/LinkedList.java]: This is the wrong approach.  You want these to remain private and for the Stack and Queue classes to only use the public methods of this class.

File: src/lists/StacksAndQueues/Queue.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    private LinkedList mainStack = new LinkedList();
+    
+    @Override
+    public boolean isEmpty() {
+        return mainStack.isEmpty();
+    }
+    
+    @Override
+    public String pop() {
+        String returnValue = mainStack.head.value;
```
Comment (booleangate) [inline src/lists/StacksAndQueues/Queue.java]: What happens if someone calls `pop()` when the list is empty?

File: src/lists/StacksAndQueues/Queue.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        
+        if (mainStack.size() > 0) {
```
Comment (booleangate) [inline src/lists/StacksAndQueues/Queue.java]: Since the `List` interface has the `isEmpty()` method, we should use that instead of comparing  `size()` against `0`: `if(!mainStack.isEmpty())`

The reason for this is readability.  Both statements are the same, but the _intention_ of "if list not empty" is clearer than "if list size is greater than 0."

File: src/lists/LinkedList.java (hunk 1 above)
Comment (r3ddawn) [inline src/lists/LinkedList.java]: Realized I could use .get() and .delete() to do what I needed rather than doing it manually. This allowed me to keep Node private.

File: src/lists/StacksAndQueues/Stack.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    @Override
+    public String pop() {
+        String temp = mainQueue.get(0);
+        mainQueue.delete(0);
+        return temp;
+    }
+    
+    @Override
+    public void push(String value) {
+        if (mainQueue.size() > 0) {
```
Comment (booleangate) [inline src/lists/StacksAndQueues/Stack.java]: Use`isEmpty()` here.

File: test/StackTest.java [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+         stack.push("Hello");
+         stack.push("Goodbye");
+         stack.push("Pop Me");
+
+         stack.print();
+     }
+     
+     @Test
+     public void pushAndPop() {
+        Stack stack = new Stack();
```
Comment (booleangate) [inline test/StackTest.java]: Add a test to `pop()` from an empty stack.

File: test/QueueTest.java [hunk 7]
```diff
@@ -0,0 +1,24 @@
+import lists.StacksAndQueues.Queue;
+import static org.junit.Assert.assertEquals;
+import org.junit.Test;
+
+/**
+ *
+ * @author ericjohnson
+ */
+public class QueueTest {
```
Comment (booleangate) [inline test/QueueTest.java]: Add a test to `pop()` from an empty queue.

File: test/StackTest.java [hunk 8]
```diff
@@ -0,0 +1,36 @@
+import org.junit.Test;
+import lists.StacksAndQueues.Stack;
+import static org.junit.Assert.assertEquals;
+
+/**
+ *
+ * @author ericjohnson
+ */
+public class StackTest {
```
Comment (booleangate) [inline test/StackTest.java]: Add a test to `pop()` from an empty stack.
